src/coreclr/jit/assertionprop.cpp

```diff
@@ -1136,16 +1143,18 @@ AssertionIndex Compiler::optCreateAssertion(GenTree*         op1,
                         assert(op2->IsIntegralConst(0));
                         op2Kind = O2K_ZEROOBJ;
                     }
+#ifndef TARGET_64BIT
+                    else if (op1->TypeIs(TYP_LONG))
```

Comment:
What's the reason for needing `O2K_CONST_LONG` on 32-bit, but not on 64-bit where we can also be `TYP_LONG`?